lib/api/CInferenceModelDefinition.cc

```diff
@@ -196,6 +223,26 @@ CTree::TTreeNodeVec& CTree::treeStructure() {
     return m_TreeStructure;
 }
 
+CTrainedModel::TStringVec CTree::removeUnusedFeatures() {
+    boost::unordered_map<std::size_t, std::size_t> selectedFeatureIndices;
+    for (auto& treeNode : m_TreeStructure) {
+        if (treeNode.leaf() == false) {
+            std::size_t adjustedIndex{selectedFeatureIndices
+                                          .emplace(treeNode.splitFeature(),
+                                                   selectedFeatureIndices.size())
+                                          .first->second};
+            treeNode.splitFeature(adjustedIndex);
+        }
+    }
+    TStringVec selectedFeatureNames(selectedFeatureIndices.size());
+    auto& featureNames = this->featureNames();
+    for (auto i = selectedFeatureIndices.begin(); i != selectedFeatureIndices.end(); ++i) {
+        selectedFeatureNames[i->second] = std::move(featureNames[i->first]);
+    }
+    this->featureNames(std::move(selectedFeatureNames));
+    return selectedFeatureNames;
```

Comment:
You're returning after moving from this type in the line above.

Reply:
Fixed. Thank you for catching this. :+1: